### ISSUE TITLE
Can not automatically select forced subtitles, --sub-forced-only=yes makes subtitles not render

### ISSUE THREAD

**Issue by Ramog** [mpv-player/mpv#13804]
### Important Information
I have tried a few flags (--sub-forced-only=yes, --subs-fallback-forced=always) and I can't get it to work. Furthermore --sub-forced-only=yes or --sub-forced-only makes it so no subtitle gets rendered at all regardless of language and forced-flag.

It also seems like the first english subtitle track is always jumped over if there are multiples (in this file 3), regardless of it has a forced tag or not. If I delete all other tracks then the first is choosen (even if its a forced track).

Provide following Information:
- mpv version: 0.37.0-709
- Windows Version: win10
- Source of the mpv binary: https://sourceforge.net/projects/mpv-player-windows/files/64bit-v3/ -> [mpv-x86_64-v3-20240331-git-765a43a.7z](https://sourceforge.net/projects/mpv-player-windows/files/64bit-v3/mpv-x86_64-v3-20240331-git-765a43a.7z/download)
- If known which version of mpv introduced the problem: not known
- GPU model, driver and version: RTX 4070, 551.86
- Screenshot of MKVToolNix GUI of the file I tried to play: 
![grafik](https://github.com/mpv-player/mpv/assets/14116991/5ba3bc09-c7a4-4d3e-b573-a5e2e9e6eca5)
- In the case of the 2nd subtitle being the forced one it selects it (but thats just because it always selects the second one)
![grafik](https://github.com/mpv-player/mpv/assets/14116991/089704ec-d2ec-4024-ad57-b53fdc4982bf)
- Same exact settings just the the forced track being swapped with the normal track (it still selects 2nd one)
![grafik](https://github.com/mpv-player/mpv/assets/14116991/234f891c-27da-4f72-8e09-4aa948995421)
- If I start MPV with --subs-fallback-forced subtitles don't rendered at all
![grafik](https://github.com/mpv-player/mpv/assets/14116991/43c78750-5fcb-4120-ab21-253d1399da1c)

If you're not using git master or the latest release, update.
Releases are listed here: https://github.com/mpv-player/mpv/releases

### Reproduction steps
Start mpv over cli with --sub-forced-only=yes flag, get a file ready that has an english subtitle track with flags and another english subtitle track and drag it into the player.

### Expected behavior
Mpv automatically choosing and displying the right english track that has the forced flag.

### Actual behavior
- Mpv selects the second english subtitle track all the time, while complelty ignoring the forced flag.
- When using  --sub-forced-only flag the subtitles don't get rendered at all-
### Log file
[output.txt](https://github.com/mpv-player/mpv/files/14819449/output.txt)

**Comment by Dudemanguy** [comment]
Can you share a sample of this? I think there's an edge case not being handled here since I can't reproduce on my own samples.

>Furthermore --sub-forced-only=yes or --sub-forced-only makes it so no subtitle gets rendered at all regardless of language and forced-flag.

That's expected. This option doesn't do what you think it does.

**Comment by Ramog** [comment]
@Dudemanguy 
> Can you share a sample of this? I think there's an edge case not being handled here since I can't reproduce on my own samples.

Sure I have made them as small as possible and contain almost nothing except the structure that causes the issue (mostly black video, and silent audio track all in same format) (I tested them in MPV they still select the second englisch track. In VLC the forced one gets always selected ). Apologies for the weird resolution ratio, it was the only way I could think of right now.
Had to put them into a zipfile because mkv of all things is not on the allowed file type list (mov and mp4 would be xD).
[Filesamples.zip](https://github.com/mpv-player/mpv/files/14826861/Filesamples.zip)
> > Furthermore --sub-forced-only=yes or --sub-forced-only makes it so no subtitle gets rendered at all regardless of language and forced-flag.
> 
> That's expected. This option doesn't do what you think it does.

Then what does it do? Ubuntu manual (yes I am on windows 10 but I needed a source for the cli flags) it states this
```
_--sub-forced-only_
 Display only forced  subtitles  for  the  DVD  subtitle  stream  selected  by  e.g.
--slang.
```
-- https://manpages.ubuntu.com/manpages/xenial/en/man1/mpv.1.html
With a quick research I found out that even on a DVD the forced subtitles are typically saved on a track thats marked forced and not in the same track as other subtitles so I don't get what the flag would do if not what I am trying to do ^^'

**Comment by Dudemanguy** [comment]
It's a bad name and that manual entry is deceiving. I renamed it to `--sub-forced-events-only` a while ago and changed the manual description to be a little clearer. It currently says:

```
--sub-forced-events-only=<yes|no>
              Enabling this displays only forced events within subtitle streams. Only some bitmap subtitle formats (such as DVD or PGS) are capable of having a mixture of forced and unforced events within
              the stream. Enabling this on text subtitles will cause no subtitles to be displayed (default: no).
```

The option only has meaning for bitmap subtitle formats and only if the stream mixes forced and unforced events (this is rare).

Anyways thanks for the sample. The problem is what I thought it was.

**Comment by Ramog** [comment]
Okay well that explains why the setting doesn't do anything but even then its still choosing the wrong subtitle track doesn't it? Or does mpv just not support forced subtitle tracks being choosen over normal ones?

**Comment by Dudemanguy** [comment]
>And why does it stick to the 2nd subtitle track out of 3

Coincidence. It was correctly picking forced subs when it was the second track but not the first track. I fixed it in the above commit.

**Comment by Ramog** [comment]
> > And why does it stick to the 2nd subtitle track out of 3
> 
> Coincidence. It was correctly picking forced subs when it was the second track but not the first track. I fixed it in the above commit.

Oh okay, sorry, didn't understand completely what was going on before, thought this was marked as closed and completed and was confused if the other issues were forgotten, thank you for explaining it to me ^-^

**Comment by ccheath** [comment]
I recently ran upon this after distro upgrade.
I had previously had `subbed-forced-only=yes` in my config.
I made the addition to my config a while back when trying to figure out how to get forced subs to show automatically.
I thought they would show automatically by default (ya know, 'forced'?) but they weren't.
Adding the `subbed-forced-only=yes` made it so that the forced subs would always render.
After the distro upgrade they were no longer rendering (and no subs were rendering at all).
... and after some searching online, well here I am.

I've removed that line from my config and now subs are working again, but I have to manually select the forced subs to get them to render.

**Is there something else that I need to add to my config to get the forced subs to render automatically?**

PS: I'm actually running celluloid on Linux Mint 22 (wilma).
TYIA
